rspace/src/main/scala/coop/rchain/rspace/history/HistoryRepositoryImpl.scala

```diff
@@ -129,44 +116,63 @@ final case class HistoryRepositoryImpl[F[_]: Sync: Parallel, C, P, A, K](
   private def computeMeasure(actions: List[HotStoreAction]): List[String] =
     actions.map {
       case i: InsertData[C, A] =>
-        val key  = hashDataChannel(i.channel).bytes
+        val key  = hashDataChannel(i.channel, codecC).bytes
         val data = encodeData(i.data)
         s"${key.toHex};insert-data;${data.length};${i.data.length}"
       case i: InsertContinuations[C, P, K] =>
-        val key  = hashContinuationsChannels(i.channels).bytes
+        val key  = hashContinuationsChannels(i.channels, serializeC).bytes
         val data = encodeContinuations(i.continuations)
         s"${key.toHex};insert-continuation;${data.length};${i.continuations.length}"
       case i: InsertJoins[C] =>
-        val key  = hashJoinsChannel(i.channel).bytes
+        val key  = hashJoinsChannel(i.channel, codecC).bytes
         val data = encodeJoins(i.joins)
         s"${key.toHex};insert-join;${data.length}"
       case d: DeleteData[C] =>
-        val key = hashDataChannel(d.channel).bytes
+        val key = hashDataChannel(d.channel, codecC).bytes
         s"${key.toHex};delete-data;0"
       case d: DeleteContinuations[C] =>
-        val key = hashContinuationsChannels(d.channels).bytes
+        val key = hashContinuationsChannels(d.channels, serializeC).bytes
         s"${key.toHex};delete-continuation;0"
       case d: DeleteJoins[C] =>
-        val key = hashJoinsChannel(d.channel).bytes
+        val key = hashJoinsChannel(d.channel, codecC).bytes
         s"${key.toHex};delete-join;0"
     }
 
-  private def transformAndStore(hotStoreActions: List[HotStoreAction]): F[List[HistoryAction]] =
+  private def transformAndStore(hotStoreActions: List[HotStoreAction]): F[List[HistoryAction]] = {
+    import cats.instances.list._
     for {
-      transformedActions <- Sync[F].delay(hotStoreActions.map(transform))
+      transformedActions <- hotStoreActions.map(transform).pure[F]
```

Comment:
This change will brake referential transparency for this function. `hotStoreActions.map(transform)` will be invoked when `transformAndStore` is called and this is not what is expected.

With `Sync[F].delay(...)` execution will be _delayed_ until the whole F expression is executed.

Reply:
Cool. Great to know that! But how does this break `brake referential transparency for this function`?